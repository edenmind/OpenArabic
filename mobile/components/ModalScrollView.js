import PropTypes from 'prop-types'
import React from 'react'
import { ScrollView } from 'react-native-gesture-handler'
import { Button, Modal, Portal, Text } from 'react-native-paper'
import { COLORS } from '../constants/colors'

const ModalScrollView = (props) => {
  const containerStyle = {
    backgroundColor: COLORS.shinyOlive,
    margin: 20,
    padding: 20
  }

  const buttonPadding = {
    paddingTop: 30
  }

  return (
    <Portal>
      <Modal
        visible={props.visible}
        onDismiss={props.hideModal}
        contentContainerStyle={containerStyle}>
        <ScrollView>{props.content}</ScrollView>
        <Button onPress={props.hideModal} style={buttonPadding}>
          <Text>Close</Text>
        </Button>
      </Modal>
    </Portal>
  )
}

export default ModalScrollView

ModalScrollView.propTypes = {
  content: PropTypes.any.isRequired,
  visible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired
}
