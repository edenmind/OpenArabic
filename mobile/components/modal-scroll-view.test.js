import { describe, it, jest } from '@jest/globals'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6
import React from 'react'
import ModalScrollView from './modal-scroll-view.js'
import renderer from 'react-test-renderer'

jest.useFakeTimers()
describe('<ModalScrollView />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ModalScrollView title="abc" hideModal={() => {}} visible={true} content={'content'} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    // @ts-ignore
    const renderer = new ShallowRenderer()
    renderer.render(<ModalScrollView title="abc" hideModal={() => {}} visible={true} content={'content'} />)
  })
})
