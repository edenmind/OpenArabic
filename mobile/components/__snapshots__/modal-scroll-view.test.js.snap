// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalScrollView /> renders correctly 1`] = `null`;
